### PR TITLE
Conda build

### PR DESCRIPTION
**Pull request by shinaoka**
Build conda packages for a product of (Python 3.6, 3.7, 3.8, 3.9), (numpy 1.19, 1.20, 1.20), (Linux x64, Mac x64, Win x64).
If you do not like the messy commit messages, we can use squash merge. The optimization options "-march=native, -mtune=native" are disabled. Should we introduce an environment variable to activate them?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -107,8 +107,11 @@ def build_extensions(self):
             compiler_make = 'msvc'
 
         if compiler_type != 'msvc':
-            new_flags = {"-Wextra": None, "-std": "c11",
-                         "-march": "native", "-mtune": "native"}
+            # Prevent generation of code for the compiling machine (for the moment)
+            # This is harmful when building a binary package.
+            # new_flags = {"-Wextra": None, "-std": "c11",
+            #       "-march": "native", "-mtune": "native"}
+            new_flags = {"-Wextra": None, "-std": "c11"}
```
Comment (mwallerb) [inline setup.py:114]: Can you add a new option for adding this back (see `user_options` above)?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+RUN useradd -u $NB_UID -m $NB_USER -s /bin/bash && \
+    echo 'vscode ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+USER $NB_USER
+ENV PATH "/home/${NB_USER}/.local/bin:${PATH}"
+ENV PYTHONPATH "/home/${NB_USER}/work/src:${PYTONPATH}"
+
+# for vscode
+RUN mkdir /home/${NB_USER}/work
+
+RUN conda config --add pkgs_dirs /home/vscode/.conda/pkgs
```
Comment (mwallerb) [inline .devcontainer/Dockerfile]: I don't understand what this `.devcontainer` does ... is this necessary for anaconda deployment somehow?

**Review by mwallerb — COMMENTED** [review]
Looks good!  Couple of minor things ...

File: setup.py (hunk 1 above)
Comment (shinaoka) [reply setup.py:114]: I have no idea how to introduce a new option systematically.
Should we introduce a command-line option to `setup.py`?
Or, how about checking an environment variable `XPREC_NATIVE_OPT`?

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (shinaoka) [reply .devcontainer/Dockerfile]: I will hide it in `dev/`.

**Pull request merged**